src/app/TypeormStore/TypeormStore.ts

```diff
@@ -130,21 +130,33 @@ export class TypeormStore extends Store {
         )
       : Promise.resolve()
     )
-      .then(() =>
-        this.repository.save({
-          expiredAt: Date.now() + ttl * 1000,
-          id: sid,
-          json,
-        }),
-      )
+      // @ts-ignore
+      .then(async () => {
+        try {
```

Comment:
the new logic seems very different but looking at the actual queries is very similar.

repository.save does a select, then decides to do an update vs an insert.

the most important thing, however, is to add a condition that `destroyedAt IS NULL` in the update. typeorm soft delete doesn't add this clause by default, unfortunately. Without this condition, during the race condition the session is updated (in the test above, views becomes 3).